decentraland/unity-renderer
# feat: add feature flag for animation culling setting

## What does this PR change?
Issue #2655

We had legacy code that was disabling animation culling, as a temporary fix for a bug.
![image (6)](https://user-images.githubusercontent.com/7305682/179704772-67f02047-8541-493b-8f01-8db3d60d620e.png)
That logic was setting in animation components `cullingType = AlwaysAnimate`

This PR is introducing feature flag, where we can control the setting, to test the performance impact.
I have also added debug option to test switching the setting during runtime.

## How to test the changes?
Change [flag ](https://features.decentraland.systems/#/features/strategies/explorer-animation_culling_status) value.
- When flag is set to `false`
  - Default Behaviour, as we have right now
  - In editor, the animation components should have `cullingType` set to `AlwaysAnimate`.

- When flag is set to `true`
  - In editor, the animation components should have `cullingType` set to `Based On Renderer`.

In both scenarios, there should not be any visible change in animations.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
